kinto/core/authentication.py

```diff
@@ -37,6 +44,93 @@ def unauthenticated_userid(self, request):
             return userid
 
 
+@implementer(IAuthenticationPolicy)
+class OpenIDConnectPolicy(base_auth.CallbackAuthenticationPolicy):
+    def __init__(self, realm='Realm'):
+        self.realm = realm
+        self._cache = None
+
+        self._jwt_keys = None
+
+    def unauthenticated_userid(self, request):
+        """Return the FxA userid or ``None`` if token could not be verified.
+        """
+        authorization = request.headers.get('Authorization', '')
+        try:
+            authmeth, payload = authorization.split(' ', 1)
+        except ValueError:
+            return None
+
+        if authmeth.lower() != 'bearer+oidc':
+            return None
+
+        try:
+            # Bearer+OIDC id_token=jwt, access_token=bearer
```

Comment:
Out of curiosity, is this approach based on some prior art, or your own scheme?  I'm not sure how well it will interact with refresh tokens, which would allow you to generate a fresh access_token that's not bound to the id_token.

Reply:
It is our own scheme. We asked ourselves a lot of question about that.

Looking at the Auth0 documentation is seems that the only way to verify an access_token is to use the id_token. (at_hash value) but it doesn't seems to be the way of doing it with Firefox Account where we would most likely use the `/verify` endpoint.

Reply:
My understanding of OAuth was that sending the access_token should be enough but then how do you validate it with Auth0? How does services usually work with id_token and access_token ? Do they send the id_token?

Reply:
> My understanding of OAuth was that sending the access_token should be enough

That's my understanding as well.  The id_token is a point-in-time confirmation that we successfully *authenticated*  a particular user, and the access_token is an ongoing grant of *authorization* to do a particular thing.  I don't know much about Auth0, but having to send both on every request doesn't mesh with my expectations of OIDC.

Reply:
> Looking at the Auth0 documentation is seems that the only way to verify an access_token is to use the id_token

Can you please drop a link to this for my reference?

Reply:
This was it: https://auth0.com/docs/api-auth/tutorials/verify-access-token

But this is what I also found: https://auth0.com/docs/api-auth/intro#calling-apis-with-access-tokens

Reply:
So maybe the access_token is also a JWT that self validates?

Reply:
The problem is that if you look at this table: https://auth0.com/docs/api-auth/intro#how-to-use-the-new-flows

You see that tokeninfo is disabled...

Reply:
Should we rely on userinfo call instead?